api/services/execution/execution.proto

```diff
@@ -34,6 +36,7 @@ message CreateRequest {
 	string stdout = 6;
 	string stderr = 7;
 	bool terminal = 8;
+	string checkpoint = 9 [(gogoproto.customtype) = "github.com/opencontainers/go-digest.Digest", (gogoproto.nullable) = false];
```

Comment:
We may want to have the whole descriptor type to allow versioning and type safety of the checkpoints.